app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,36 @@ def index
 
   def show
     @post = Post.find(params[:id])
+    @user = User.find(params[:user_id])
+  end
+
+  def new
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author_id = current_user.id
+    respond_to do |format|
```

Comment:
## OPTIONAL
* I strongly suggest you add a `post_params ` method, This is a better security practice to help prevent accidentally allowing users to update sensitive model attributes. [Read more here about strong params](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)

Reply:
I'm applying your suggestion tomorrow😄